pkg/api/types.go

```diff
@@ -398,6 +402,22 @@ type NFSVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
+type GlusterfsVolumeSource struct {
+	// Required: EndpointsName is the endpoint name that details Glusterfs topology
+	EndpointsName string `json:"endpoints"`
+
+	// Required: Path is the Glusterfs volume path
+	Path string `json:"path"`
+
+	// Optional: Defaults to false (read/write). ReadOnly here will force
+	// the Glusterfs to be mounted with read-only permissions
+	ReadOnly bool `json:"readOnly,omitempty"`
+
+	// Optional: helper is the helper utility executed prior to mount command
+	Helper string `json:"helper,omitempty"`
```

Comment:
As @erictune mentioned earlier, can you get rid of "Helper" for now? If there is a real need, we can add it in the future.